src/core/structures/PDFAcroForm.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  PDFArray,
+  PDFBool,
+  PDFContext,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFObject,
+  PDFStream,
+  PDFString,
+} from 'src/core';
+
+class PDFAcroForm extends PDFDict {
+  static fromMapWithContext(
+    dict: Map<PDFName, PDFObject>,
+    context: PDFContext,
+  ): PDFAcroForm {
+    dict.set(

```

Comment:
This is a stylistic nit-pick, but I think an explicit existence check would be a little more clear. Also, explicit construction of PDF primitive objects is discouraged in favor of the `context.obj(...)` helper, which allows you to construct PDF objects with literal syntax:

https://github.com/Hopding/pdf-lib/blob/620747a044c2b3864c9d844b7c67b924b4f94ea5/src/core/PDFContext.ts#L141-L146

For example:
```js
if (!dict.has(PDFName.of('Fields'))) {
  dict.set(PDFName.of('Fields'), context.obj([]))
}
```

Reply:
Updated